# Casusopdracht - Films

**Klas:** V2B
**Studenten:** 
- Roan Gaasbeek
- Mathijs
- Luc

### Inhoud ###
- [Onderzoeksvragen](1.-Onderzoeksvragen)

In [7]:
import numpy as np
import pandas as pd

# 1. Onderzoeksvragen
De volgende onderzoeksvragen worden behandeld in deze opdracht:

- In hoeverre is de omzet van een film te voorspellen op basis van de populariteit op $Facebook$ en $IMDB$?
- ** zelf nog een vraag bedenken met betrekking op een zelf bijgezochte dataset **
- In hoeverre is het mogelijk om logische clusters te vinden o.b.v. onder andere budget en omzet? Denk aan blockbusters (hoog budget en hoge omzet), flops (hoog budget en lage omzet) of cultfilms (laag budget en hoge omzet). Je mag extra features toevoegen. 

# 2. Data understanding

De eesrte stap in deze opdracht is de data understanding fase van $CRISP-DM$ dit houdt in:
- Data Collection
- Data Exploration & Analysis

#### 2.1 Data Collection
- de meegekregen movie.csv
- nog een dataset erbij zoeken

In [8]:
# Dit is de casus dataset
dataset = pd.read_csv('movie.csv')

# dit is de gekozen externe dataset


#### 2.2 Data Exploration & Analysis

In [9]:
# nu gebruiken we deze functie om een eerste blik van de dataset te krijgen
dataset.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4    NaN        Doug Walker                     NaN       NaN   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    131.0                     NaN        Rob Walker   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   131.0          NaN                      Documentary  ...   

  num_user_for_reviews language  country  content_rating       budget  \
0               3054.0  English      USA           PG-13  237000000.0   
1               1238.0  English      USA           PG-13  300000000.0   
2                994.0  English       UK           PG-13  245000000.0   
3               2701.0  English      USA           PG-13  250000000.0   
4                  NaN      NaN      NaN             NaN          NaN   

   title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0      2009.0                  936.0        7.9          1.78   
1      2007.0                 5000.0        7.1          2.35   
2      2015.0                  393.0        6.8          2.35   
3      2012.0                23000.0        8.5          2.35   
4         NaN                   12.0        7.1           NaN   

  movie_facebook_likes  
0                33000  
1                    0  
2                85000  
3               164000  
4                    0  

[5 rows x 28 columns]

In [10]:
# dit zijn de features met de datatypen
dataset.dtypes

color                         object
director_name                 object
num_critic_for_reviews       float64
duration                     float64
director_facebook_likes      float64
actor_3_facebook_likes       float64
actor_2_name                  object
actor_1_facebook_likes       float64
gross                        float64
genres                        object
actor_1_name                  object
movie_title                   object
num_voted_users                int64
cast_total_facebook_likes      int64
actor_3_name                  object
facenumber_in_poster         float64
plot_keywords                 object
movie_imdb_link               object
num_user_for_reviews         float64
language                      object
country                       object
content_rating                object
budget                       float64
title_year                   float64
actor_2_facebook_likes       float64
imdb_score                   float64
aspect_ratio                 float64
m

met de volgende functie krijgen we een mooie samenvatting van de data

In [11]:
dataset.describe()

num_critic_for_reviews     duration  director_facebook_likes  \
count             4993.000000  5028.000000              4939.000000   
mean               140.194272   107.201074               686.509212   
std                121.601675    25.197441              2813.328607   
min                  1.000000     7.000000                 0.000000   
25%                 50.000000    93.000000                 7.000000   
50%                110.000000   103.000000                49.000000   
75%                195.000000   118.000000               194.500000   
max                813.000000   511.000000             23000.000000   

       actor_3_facebook_likes  actor_1_facebook_likes         gross  \
count             5020.000000             5036.000000  4.159000e+03   
mean               645.009761             6560.047061  4.846841e+07   
std               1665.041728            15020.759120  6.845299e+07   
min                  0.000000                0.000000  1.620000e+02   
25%                133.000000              614.000000  5.340988e+06   
50%                371.500000              988.000000  2.551750e+07   
75%                636.000000            11000.000000  6.230944e+07   
max              23000.000000           640000.000000  7.605058e+08   

       num_voted_users  cast_total_facebook_likes  facenumber_in_poster  \
count     5.043000e+03                5043.000000           5030.000000   
mean      8.366816e+04                9699.063851              1.371173   
std       1.384853e+05               18163.799124              2.013576   
min       5.000000e+00                   0.000000              0.000000   
25%       8.593500e+03                1411.000000              0.000000   
50%       3.435900e+04                3090.000000              1.000000   
75%       9.630900e+04               13756.500000              2.000000   
max       1.689764e+06              656730.000000             43.000000   

       num_user_for_reviews        budget   title_year  \
count           5022.000000  4.551000e+03  4935.000000   
mean             272.770808  3.975262e+07  2002.470517   
std              377.982886  2.061149e+08    12.474599   
min                1.000000  2.180000e+02  1916.000000   
25%               65.000000  6.000000e+06  1999.000000   
50%              156.000000  2.000000e+07  2005.000000   
75%              326.000000  4.500000e+07  2011.000000   
max             5060.000000  1.221550e+10  2016.000000   

       actor_2_facebook_likes   imdb_score  aspect_ratio  movie_facebook_likes  
count             5030.000000  5043.000000   4714.000000           5043.000000  
mean              1651.754473     6.442138      2.220403           7525.964505  
std               4042.438863     1.125116      1.385113          19320.445110  
min                  0.000000     1.600000      1.180000              0.000000  
25%                281.000000     5.800000      1.850000              0.000000  
50%                595.000000     6.600000      2.350000            166.000000  
75%                918.000000     7.200000      2.350000           3000.000000  
max             137000.000000     9.500000     16.000000         349000.000000

een paar dingen die opvallen zijn:
- veel data die een minimum waarde van 0 hebben zijn dit missende waardes of daadwerkelijk 0?
- 

### Analyse van feature- en targetvariabelen

Te onderzoeken featurevariabelen:  
- movie_facebook_likes
- imdb_score
- budget

Te onderzoeken targetvariabele:  
- gross

In [15]:
# Kolommen selecteren
kolommen = ['movie_facebook_likes', 'imdb_score', 'budget', 'gross']
data_analyse = dataset[kolommen]
data_analyse


movie_facebook_likes  imdb_score       budget        gross
0                    33000         7.9  237000000.0  760505847.0
1                        0         7.1  300000000.0  309404152.0
2                    85000         6.8  245000000.0  200074175.0
3                   164000         8.5  250000000.0  448130642.0
4                        0         7.1          NaN          NaN
...                    ...         ...          ...          ...
5038                    84         7.7          NaN          NaN
5039                 32000         7.5          NaN          NaN
5040                    16         6.3       1400.0          NaN
5041                   660         6.3          NaN      10443.0
5042                   456         6.6       1100.0      85222.0

[5043 rows x 4 columns]

In [17]:
# Centrum- en spreidingsmaten
data_analyse.describe()

movie_facebook_likes   imdb_score        budget         gross
count           5043.000000  5043.000000  4.551000e+03  4.159000e+03
mean            7525.964505     6.442138  3.975262e+07  4.846841e+07
std            19320.445110     1.125116  2.061149e+08  6.845299e+07
min                0.000000     1.600000  2.180000e+02  1.620000e+02
25%                0.000000     5.800000  6.000000e+06  5.340988e+06
50%              166.000000     6.600000  2.000000e+07  2.551750e+07
75%             3000.000000     7.200000  4.500000e+07  6.230944e+07
max           349000.000000     9.500000  1.221550e+10  7.605058e+08

### Waarnemingen centrum- en spreidingsmaten

- De kolommen movie_facebook_likes en imdb_score hebben beide 5043 waarden, maar budget en gross hebben minder waarden (4551 en 4159 waarden).  
We zullen hier een oplossing voor moeten vinden, door bijvoorbeeld de rijen met missende waarden te droppen van de dataset.
- Q1 voor movie_facebook_likes is 0, dit betekent dat ten minste 25% van de films 0 facebook likes hebben. Dit komt waarschijnlijk omdat deze films geen facebook-pagina hebben.


In [ ]:
# Outliers bepalen

# 3. Data preparation
De volgende stap in deze is de data preparation fase van $CRISP-DM$ dit houdt in:
- Data Cleaning
- Data Processing

#### 3.1 Data Processing

{hier kunnen we colummen die we niet nodig hebben verwijderen en de volgorde aanpassen}


In [12]:
dataset.drop(columns=['color','director_name','director_facebook_likes','actor_1_name', 'actor_1_facebook_likes','actor_2_name', 'actor_2_facebook_likes', 'actor_3_name', 'actor_3_facebook_likes','duration','genres','num_voted_users','cast_total_facebook_likes','facenumber_in_poster','plot_keywords', 'num_user_for_reviews','num_critic_for_reviews', 'language', 'country', 'content_rating', 'title_year','aspect_ratio'], inplace=True)

dataset.rename(columns={'movie_title':'title'}, inplace=True)

dataset.head()

gross                                              title  \
0  760505847.0                                            Avatar    
1  309404152.0          Pirates of the Caribbean: At World's End    
2  200074175.0                                           Spectre    
3  448130642.0                             The Dark Knight Rises    
4          NaN  Star Wars: Episode VII - The Force Awakens    ...   

                                     movie_imdb_link       budget  imdb_score  \
0  http://www.imdb.com/title/tt0499549/?ref_=fn_t...  237000000.0         7.9   
1  http://www.imdb.com/title/tt0449088/?ref_=fn_t...  300000000.0         7.1   
2  http://www.imdb.com/title/tt2379713/?ref_=fn_t...  245000000.0         6.8   
3  http://www.imdb.com/title/tt1345836/?ref_=fn_t...  250000000.0         8.5   
4  http://www.imdb.com/title/tt5289954/?ref_=fn_t...          NaN         7.1   

   movie_facebook_likes  
0                 33000  
1                     0  
2                 85000  
3                164000  
4                     0

** mischien dat we de facebook likes willen bewaren maar wilde gewoon wat proberen **

#### 3.2 Data Cleaning

